test-framework/emulator_backend.go

```diff
@@ -277,6 +281,85 @@ func (e *EmulatorBackend) CommitBlock() error {
 	return err
 }
 
+func (e *EmulatorBackend) DeployContract(
+	name string,
+	code string,
+	account *interpreter.Account,
+	args []interpreter.Value,
+) error {
+
+	const deployContractTransactionTemplate = `
+	    transaction(%s) {
+		    prepare(signer: AuthAccount) {
+			    signer.contracts.add(name: "%s", code: "%s".decodeHex()%s)
+		    }
+	    }`
+
+	hexEncodedCode := hex.EncodeToString([]byte(code))
+
+	inter, err := newInterpreter()
+	if err != nil {
+		return err
+	}
+
+	cadenceArgs := make([]cadence.Value, 0, len(args))
+
+	txArgs, addArgs := "", ""
+
+	for i, arg := range args {
+		cadenceArg, err := runtime.ExportValue(arg, inter, interpreter.ReturnEmptyLocationRange)
+		if err != nil {
+			return err
+		}
+
+		if i > 0 {
+			txArgs += ", "
+		}
+
+		txArgs += fmt.Sprintf("arg%d: %s", i, cadenceArg.Type().ID())
+		addArgs += fmt.Sprintf(", arg%d", i)
```

Comment:
Use string builders instead of repeated string concatenation

Reply:
fixed in 07288813e3a2d990e8637b2a02415c4396b0af7c